.github/workflows/ci.yml

```diff
@@ -57,43 +71,32 @@ jobs:
            pip install tqdm --break-system-packages
            make bats-nocontainer
 
-  docker:
+  bats-nocontainer:
     runs-on: ubuntu-24.04
     steps:
       - uses: actions/checkout@v4
       - name: install bats
         shell: bash
         run: |
            sudo apt-get update
-           sudo apt-get install bats bash codespell python3-argcomplete pipx
+           sudo apt-get install podman bats bash codespell python3-argcomplete pipx
            make install-requirements
-      - name: Print disk space before cleanup
-        shell: bash
-        run: |
-           df -h
-      - name: Free Disk Space Linux
-        shell: bash
-        run: |
-           sudo docker rmi "$(docker image ls -aq)" >/dev/null 2>&1 || true
-           sudo rm -rf \
-              /usr/share/dotnet /usr/local/lib/android /opt/ghc \
-              /usr/local/share/powershell /usr/share/swift /usr/local/.ghcup \
-              /usr/lib/jvm || true
 
-      # /mnt has ~ 65 GB free disk space. / is too small.
-      - name: Reconfigure Docker data-root
+      - name: bats-nocontainer
         run: |
-           sudo mkdir -p /mnt/docker /etc/docker
-           echo '{"data-root": "/mnt/docker"}' > /tmp/daemon.json
-           sudo mv /tmp/daemon.json /etc/docker/daemon.json
-           cat /etc/docker/daemon.json
-           sudo systemctl restart docker.service
-           df -h
+           pip install tqdm --break-system-packages
+           make bats-nocontainer
 
-      - name: Print disk space after cleanup
+  docker:
```

Comment:
X-post from my PR
I think this job needs the container images to be built before bats-docker is ran, within the same job.  Otherwise we will be pulling container images from the registry (and CUDA images will fail).  

This means we are not testing changes from the PR, but rather always testing against the images built from the main branch.

Reply:
Discussed in the other PR so closing this